var/spack/repos/builtin/packages/py-your/package.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyYour(PythonPackage):
+    """Python library to read and process pulsar data in several different formats"""
+
+    homepage = "https://github.com/thepetabyteproject/your"
+
+    # pypi tarball has requirements.txt missing
+    url = "https://github.com/thepetabyteproject/your/archive/refs/tags/0.6.7.tar.gz"
+
+    maintainers("aweaver1fandm")
+
+    license("GPL-3.0")
+
+    version("0.6.7", sha256="f2124a630d413621cce067805feb6b9c21c5c8938f41188bd89684968478d814")
+
+    depends_on("python@3.8:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
+
+    depends_on("py-astropy@4.0:", type=("build", "run"))
+    depends_on("py-matplotlib@3.2.1:", type=("build", "run"))
+    depends_on("py-numpy@1.18.4:", type=("build", "run"))
+    depends_on("py-h5py@2.10:", type=("build", "run"))
+    depends_on("py-scikit-image@0.14.2:", type=("build", "run"))
+    depends_on("py-scipy@1.3:", type=("build", "run"))
+    depends_on("py-numba@0.48:", type=("build", "run"))
+    depends_on("py-pandas@1.0.3:", type=("build", "run"))
+    depends_on("py-rich@8:", type=("build", "run"))
+
+    def setup_run_environment(self, env):
+        env.prepend_path("PATH", self.prefix.bin)
+        env.prepend_path("PATH", self.spec["python"].prefix.bin)
```

Comment:
I don't think these should be necessary.

Reply:
AFAICT The core doesn't explicitly set these variables and there are 131 packages that do.

Reply:
Yeah, but I don't see `py-pip` doing this to ensure the `pip` command is in the `PATH`, or `py-tabulate` for the `tabulate` command (to take a much smaller package in terms of loc in `package.py`). I don't think this is necessary.

```
18:43:39 wdconinc@menelaos ~/git/spack (develop *$%>) $ which tabulate
18:43:46 wdconinc@menelaos ~/git/spack (develop *$%>) $ spack load py-tabulate/6xgmssa
18:43:57 wdconinc@menelaos ~/git/spack (develop *$%>) $ which tabulate
/opt/software/linux-ubuntu24.04-skylake/gcc-13.2.0/py-tabulate-0.9.0-6xgmssa6olalms4vmt3ttfinvssnq3fh/bin/tabulate
```